File: scripts/js/generate-readme.js
import { expandSnippets } from './snippetExpander.js';

const args = process.argv.slice(2);

let addLineNumbers = true;
if (args.includes('--no-lines')) {
  addLineNumbers = false;
  args.splice(args.indexOf('--no-lines'), 1);
}

if (args.length === 0) {
  expandSnippets('.', null, addLineNumbers);
} else if (args.length === 1) {
  expandSnippets(args[0], null, addLineNumbers);
} else {
  expandSnippets(args[0], args[1], addLineNumbers);
}

File: scripts/py/generate-readme.py
import sys
sys.pycache_prefix='C:/__pycache__'
from snippet_expander import expand_snippets

if __name__ == "__main__":
    args = sys.argv[1:]

    add_line_numbers = True
    if "--no-lines" in args:
        add_line_numbers = False
        args.remove("--no-lines")

    if len(args) == 0:
        expand_snippets(add_line_numbers=add_line_numbers)
    elif len(args) == 1:
        expand_snippets(path_dir=args[0], add_line_numbers=add_line_numbers)
    elif len(args) >= 2:
        expand_snippets(path_dir=args[0], blanks_dir=args[1], add_line_numbers=add_line_numbers)
